pkg/ui/src/views/cluster/containers/timescale/index.tsx

```diff
@@ -108,8 +108,13 @@ class TimeScaleDropdown extends React.Component<TimeScaleDropdownProps, {}> {
     // If the timescale extends into the future then fallback to a default
     // timescale. Otherwise set the key to "Custom" so it appears correctly.
     if (!windowEnd || windowEnd > moment().subtract(currentScale.windowValid)) {
-      if (_.find(timewindow.availableTimeScales, { windowSize: { _data: windowSize._data } } as any)) {
-        selected = (_.find(timewindow.availableTimeScales, { windowSize: { _data: windowSize._data } } as any));
+      const size = { windowSize: { _data: windowSize._data } } ;
+      if (_.find(timewindow.availableTimeScales, size as any)) {
+        const data = {
+          ..._.find(timewindow.availableTimeScales, size as any),
```

Comment:
As a nit pick, it seems like you could capture `_.find(timewindow.availableTimeScales, size as any)` in a variable since you are referencing it twice. It might also help the readability in using the spread operator,

```js
const availableTimeScaleSize = _.find(timewindow.availableTimeScales, size as any);
const timeScaleKey = key: _.findKey(timewindow.availableTimeScales, size as any);

if (availableTimeScaleSize) {
  const data = {
    ...availableTimeScaleSize,
    key: timeScaleKey,
  };
  selected = data;
}